components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserEnvironmentVariable {
```

Comment:
It is just `EnvironmentVariable` with oneof organization_id or repository_pattern?

Reply:
there's also project en var

Reply:
Sorry I confused, I meant configuraiton_id or repository_pattern.